azurerm/resource_arm_iothub.go

```diff
@@ -332,3 +570,160 @@ func flattenIoTHubSharedAccessPolicy(input *[]devices.SharedAccessSignatureAutho
 
 	return results
 }
+
+func flattenIoTHubEndpoint(input *devices.RoutingProperties) []interface{} {
+	results := make([]interface{}, 0)
+
+	if input != nil && input.Endpoints != nil {
+
+		storageContainers := *input.Endpoints.StorageContainers
```

Comment:
there's actually a crash here if .StorageContainers is nil (since the `*` is dereferencing the pointer) - we can instead make this:

```
if container := input.Endpoints.StorageContainers; container != nil {
  # .. use *container as needed
}
```